Sources/MessagingPush/RichPush/RichPushDeliveryTracker.swift

```diff
@@ -0,0 +1,43 @@
+import CioInternalCommon
+import Foundation
+
+// sourcery: InjectRegisterShared = "RichPushDeliveryTracker"
+class RichPushDeliveryTracker {
+    let httpClient: HttpClient
+    let logger: Logger
+
+    init(httpClient: HttpClient, logger: Logger) {
+        self.httpClient = httpClient
+        self.logger = logger
+    }
+
+    func trackMetric(token: String, event: Metric, deliveryId: String, timestamp: String? = nil, onComplete: @escaping (Result<Void, HttpRequestError>) -> Void) {
```

Comment:
```suggestion
    func trackMetric(token: String, deliveryId: String, timestamp: String? = nil, onComplete: @escaping (Result<Void, HttpRequestError>) -> Void) {
```

The class name and use case seems to only be for delivered tracking. I don't think that we need to have an `event` parameter?